### PR TITLE
Proxy Kernel with named pipe support

### PR DESCRIPTION
**Pull request by TIHan**
Adds basic support for proxy kernels with a named pipe transport.

// cc @jonsequitur @colombod

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive.Tests/CompositeKernelTests.cs [hunk 1]
```diff
@@ -63,7 +64,76 @@ public async Task Handling_kernel_can_be_specified_using_kernel_name_as_a_direct
...
+            var fSharpKernel = new FSharpKernel();
+            using var kernel = new CompositeKernel
+            {
+                proxyKernel,
+                fSharpKernel
+            }.UseProxyKernel();
+            kernel.DefaultKernelName = fSharpKernel.Name;
+
+            var pipeName = Guid.NewGuid().ToString();
+            var cSharpKernel = new CSharpKernel();
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Tests/CompositeKernelTests.cs:81]: Make sure to dispose.

File: src/Microsoft.DotNet.Interactive/KernelExtensions.cs [hunk 2]
```diff
@@ -144,6 +144,39 @@ public static T UseDotNetVariableSharing<T>(this T kernel)
             return kernel;
         }
 
+        public static T UseProxyKernel<T>(this T kernel)
+            where T : CompositeKernel
+        {
+            var command = new Command("#!connect", "Connect to the specified named-pipe.")
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/KernelExtensions.cs:150]:
```suggestion
            var command = new Command("#!connect", "Connect to the specified remote kernel.")
```

**Comment by TIHan** [comment]
@jonsequitur @colombod I think this is merge-able if you want to get it in and play with it.

File: src/Microsoft.DotNet.Interactive.Tests/CompositeKernelTests.cs [hunk 3]
```diff
@@ -63,7 +64,82 @@ public async Task Handling_kernel_can_be_specified_using_kernel_name_as_a_direct
             events.Should()
                   .ContainSingle<CommandHandled>(e => e.Command == fsharpCommand);
         }
-        
+
+        [Fact]
```
[the remark this answers is not in the document]
Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive.Tests/CompositeKernelTests.cs:69]: We're working on getting rid of these timeouts.

**Comment by TIHan** [comment]
I added a new attribute: `FactSkipLinux` to handle skipping the test on linux.

File: src/Microsoft.DotNet.Interactive.Tests/CompositeKernelTests.cs [hunk 4]
```diff
@@ -63,7 +64,82 @@ public async Task Handling_kernel_can_be_specified_using_kernel_name_as_a_direct
             events.Should()
                   .ContainSingle<CommandHandled>(e => e.Command == fsharpCommand);
         }
-        
+
+        [FactSkipLinux]
+        public async Task Handling_kernel_can_be_specified_using_kernel_name_as_a_directive_as_a_proxy_named_pipe()
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Tests/CompositeKernelTests.cs:69]: I'd move these tests out of `CompositeKernelTests`.

Comment (TIHan) [reply src/Microsoft.DotNet.Interactive.Tests/CompositeKernelTests.cs]: Ok, added ProxyKernelTests

**Comment by TIHan** [comment]
Yay tests pass!

**Pull request merged**